PSReadLine/Suggestion.cs

```diff
@@ -0,0 +1,119 @@
+        private string _userInputText;
+        private bool _showSuggestion = true;
+
+        private bool IsPredictionOn => _options.PredictionSource != PredictionSource.None && _showSuggestion;
+        private bool IsPredictionOff => _options.PredictionSource == PredictionSource.None || !_showSuggestion;
+
+        /// <summary>
+        /// Turn off the prediction feature for the scope of the calling method.
+        /// Note: the calling method need to use this method with the 'using' statement/variable,
+        /// so that the suggestion feature can be properly restored.
+        /// </summary>
+        private IDisposable PredictionOff()
+        {
+            var oldSuggestionMode = _singleton._showSuggestion;
+            _singleton._showSuggestion = false;
+
+            return new Disposable(() => _singleton._showSuggestion = oldSuggestionMode);
+        }
+
+        private void ResetSuggestion()
+        {
+            _suggestionText = null;
+            _userInputText = null;
+        }
+
+        private string GetSuggestion(string text)
+        {
+            if (IsPredictionOff || LineIsMultiLine() || string.IsNullOrWhiteSpace(text))
+            {
+                ResetSuggestion();
+                return null;
+            }
+
+            try
+            {
+                if (_suggestionText == null || _suggestionText.Length <= text.Length ||
+                    text.Length < _userInputText.Length || !_suggestionText.StartsWith(text, _options.HistoryStringComparison))
+                {
+                    _userInputText = text;
+                    _suggestionText = GetHistorySuggestion(text);
+                }
+            }
+            catch
+            {
+                _suggestionText = null;
+            }
+
+            return _suggestionText?.Substring(text.Length);
+        }
+
+        /// <summary>
+        /// Accept the suggestion text if there is one.
+        /// </summary>
+        public static void AcceptSuggestion(ConsoleKeyInfo? key = null, object arg = null)
+        {
+            if (!TryGetArgAsInt(arg, out var numericArg, 1))
+            {
+                return;
+            }
+
+            if (numericArg > 0)
+            {
+                AcceptSuggestion();
+            }
+        }
+
+        private static void AcceptSuggestion()
+        {
+            if (_singleton._suggestionText != null)
+            {
+                // Ignore the visual selection.
+                _singleton._visualSelectionCommandCount = 0;
+
+                using var _ = _singleton.PredictionOff();
+                Replace(0, _singleton._buffer.Length, _singleton._suggestionText);
+            }
+        }
+
+        /// <summary>
+        /// Accept the current or the next suggestion text word.
+        /// </summary>
+        public static void AcceptNextSuggestionWord(ConsoleKeyInfo? key = null, object arg = null)
+        {
+            if (!TryGetArgAsInt(arg, out var numericArg, 1))
```

Comment:
I think I'd reject any non-null arg. It would be playing with fire to work like other args (take the nth suggestion - crazy and confusing).

You could also just ignore the arg assuming you have no ideas how you might use the arg.

Reply:
I feel it's useful to allow accepting more than 1 words from the suggestion text.

Reply:
Got it.